build.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+#  Copyright (c) 2015 - 2021, Intel Corporation
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions
+#  are met:
+#
+#      * Redistributions of source code must retain the above copyright
+#        notice, this list of conditions and the following disclaimer.
+#
+#      * Redistributions in binary form must reproduce the above copyright
+#        notice, this list of conditions and the following disclaimer in
+#        the documentation and/or other materials provided with the
+#        distribution.
+#
+#      * Neither the name of Intel Corporation nor the names of its
+#        contributors may be used to endorse or promote products derived
+#        from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+#  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+#  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+#  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+#  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+#  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+#  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+#  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+#  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+#  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+#  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+# Simple script that will build and test both the base and the service
+
+set -e
+
+# Set this variable to append configure options for base build
+# GEOPM_BASE_CONFIG_OPTIONS=
+
+# Set this variable to append configure options for service build
+# GEOPM_SERVICE_CONFIG_OPTIONS=
+
+
+if [ ! -z "$NUM_THREAD" ]; then
+    NUM_THREAD=20
+fi
+if [ ! -z "$GEOPM_SOURCE" ]; then
+    cd $GEOPM_SOURCE
+fi
+
+# Run the service build
+cd service
+./autogen.sh
+mkdir -p build
+cd build
+../configure ${GEOPM_SERVICE_CONFIG_OPTIONS}
+make -j${NUM_THREAD}
+make -j${NUM_THREAD} checkprogs
+make check
+
+# Run the base build
+cd ..
+./autogen.sh
+mkdir -p build
+cd build
+../configure --with-geopmd-lib=../service/build/.libs \
+             --with-geopmd-include=../service/src \
+             ${GEOPM_BASE_CONFIG_OPTIONS}
+make -j${NUM_THREAD}
+make -j${NUM_THREAD} checkprogs
+make check
```

Comment:
I'm surprised you didn't have to do any more hacking to your LD_LIBRARY_PATH or PYTHONPATH to make this work.

Reply:
Ah, I see now that I'm looking at the full PR that the changes in geopm_test.sh.in provide this functionality.  Nice.